backend/benefit/calculator/rules.py

```diff
@@ -282,9 +286,17 @@ def create_rows(self):
                 end_date=date_ranges[-1].end_date,
                 prefix_text="Koko ajalta",
             )
-            self._create_row(SalaryBenefitSumSubTotalsRow)
+            self._create_row(
+                SalaryBenefitSumSubTotalsRow,
+                start_date=date_ranges[0].start_date,
+                end_date=date_ranges[-1].end_date,
```

Comment:
As the date_ranges are created with get_sub_total_ranges they seem to be sorted according to the start_date values in ascending order. That makes date_ranges[0].start_date to be the minimum start_date of the date_ranges. But is date_ranges[-1].end_date always the maximum end_date of the date_ranges? Can the date ranges overlap each other?

Reply:
Actually the first start_date in the date_ranges must be equal to calculation.start_date and the last end_date must be equal to calculation.end_date, so I was able to simplify this. Added assertions about these invariants to get_sub_total_ranges.

Reply:
start_date=self.calculation.start_date,
                end_date=self.calculation.end_date,

Reply:
fixed